packages/admin-ui/src/components/Modal/state.internals.tsx

```diff
@@ -0,0 +1,28 @@
+import { useEffect } from 'react'
+import { useCallbackRef } from '../../../../admin-ui-hooks/src/useCallbackRef'
+import type { ModalStateReturn } from './state'
+
+export const kListeners = Symbol('useModalState#listeners')
```

Comment:
I don't want to expose the `kListeners` to avoid people using these details implementation, so I created this new file to be used just internally by the library. Not sure which kind of patterns do the team use to define private vs public stuff.